### PR TITLE
fix TFLOPs calculation

### PR DESCRIPTION
**Pull request by polisettyvarma**
when GQA used, we observe right TFLOPs after this fix.
when GQA is not used, huge difference in TFLOPs is solved for selective recompute granularity.
some other minor difference will also be observed as logits macs also added.

### CODE REVIEW

**Comment by polisettyvarma** [comment]
@romankap please review this

**Comment by romankap** [comment]
@polisettyvarma, you didn't add me as a reviewer, but code changes look good to me.

**Comment by polisettyvarma** [comment]
@jomayeri can you please review this

**Comment by polisettyvarma** [comment]
@tjruwase please review this.

**Review by tjruwase — APPROVED** [review]
